src/backings/structural/container.ts

```diff
@@ -58,10 +59,12 @@ export class ContainerStructuralHandler<T extends ObjectLike> extends Structural
     Object.entries(this._type.fields).forEach(([fieldName, fieldType]) => {
       try {
         if (fieldType.isBasic()) {
-          (fieldType as Type<T[keyof T]>).assertValidValue((value as T)[fieldName]);
-        } else {
           // @ts-ignore
-          fieldType.structural.assertValidValue((value as T)[fieldName]);
+          fieldType.assertValidValue(value[fieldName]);
+        }
+        if (fieldType.isComposite()) {
```

Comment:
If you wanna add another like in this code block, it will be typesafe. I have no idea how to fix wierd error with that assertValidValue. Tried every combination. I guess you had same one since it was ignored before 😄